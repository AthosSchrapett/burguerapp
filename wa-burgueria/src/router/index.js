import { createRouter, createWebHistory } from 'vue-router'

const Home = () => import("@/views/Home.vue");
const Order = () => import("@/views/Order.vue");
const Orders = () => import("@/views/Orders.vue");

const routes = [
  {
    path: '/',
    name: 'home',
    component: Home
  },
  {
    path: '/order',
    name: 'order',
    component: Order
  },
  {
    path: '/orders',
    name: 'orders',
    component: Orders
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
})

export default router
