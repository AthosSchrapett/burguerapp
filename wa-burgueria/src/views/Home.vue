<template>
  <Navbar :logo="logoUrl" :alt="app_name" @login="handleLogin" />
  <Banner />
  <div class="main-container">
    <h1>Aqui ser√° a Home Page:</h1>
  </div>
</template>

<script>
import useModal from "@/hooks/useModal";

import Navbar from "@/components/Navbar.vue";
import Banner from "@/components/Banner.vue";
export default {
  components: {
    Navbar,
    Banner,
  },
  data() {
    return {
      logoUrl: "/img/logo.png",
      app_name: "Make Your Burger",
    };
  },

  setup(){
    const modal = useModal();

    function handleLogin(){
      modal.open({
        component: "ModalLogin"
      })
    }

    return {
      handleLogin
    }
  }
};
</script>