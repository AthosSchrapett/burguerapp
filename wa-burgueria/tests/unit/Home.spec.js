import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home.vue";
import router from "@/router";

const wrapper = shallowMount(Home, {
  global: {
    plugins: [router],
  },
});

describe("<Home />", () => {
  it("should render home correctly", async () => {
    router.push("/");
    await router.isReady();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should find a div with a class main-container", async () => {
    expect(wrapper.get(".main-container"));
  });

  test("throws on octopus", () => {
    expect(() => {
      wrapper.get(".does-not-exist-class");
    }).toThrow();
  });

  it("should catch an error after does not find an nonexistent class div", async () => {
    const t = () => {
      throw new TypeError();
    };
    expect(t).toThrow(TypeError);
  });
});
