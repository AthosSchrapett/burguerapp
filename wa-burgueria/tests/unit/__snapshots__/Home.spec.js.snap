// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `
<banner-stub></banner-stub>
<div class="main-container">
  <h1>Pe√ßa seu burger:</h1>
  <burger-form-stub></burger-form-stub>
</div>
`;
